hasher-matcher-actioner/hmalib/lambdas/api/api_auth.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import os
+import json
+import time
+import jwt
+import requests
+import base64
+from jwt.algorithms import RSAAlgorithm
+
+from hmalib.common.logging import get_logger
+
+ACCESS_TOKEN = os.environ["ACCESS_TOKEN"]
+USER_POOL_URL = os.environ["USER_POOL_URL"]
+CLIENT_ID = os.environ["CLIENT_ID"]
+
+keys_url = f"{USER_POOL_URL}/.well-known/jwks.json"
+
+response = requests.get(keys_url)
+key_list = json.loads(response.text).get("keys", [])
+keys = {key["kid"]: json.dumps(key) for key in key_list}
+
+
+def is_jwt(token: str) -> bool:
+    jwt_split = token.split(".")
+    try:
+        if len(jwt_split) != 3:
+            return False
+        header = json.loads(base64.b64decode(jwt_split[0] + "=="))
+        if header["alg"] != "RS256":
+            return False
+    except Exception as e:
+        print("INFO: Provisioning Data is not in JWT format\n {0}".format(e))
+        return False
+    return True
+
+
+def validate_jwt(token: str):
+    logger = get_logger()
```

Comment:
Don't we usually put the logger at the top level because we only want to call `get_logger()` once